CMakeLists.txt

```diff
@@ -5,6 +5,19 @@ set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
+option(NOPLUGINS "Disable plugins" OFF)
```

Comment:
Would it be beneficial to add these options to the documentation or the core developer guidelines? Specifically, for the time trace functionality, I recommend providing more detailed instructions on how to run it and interpret the results. This additional information would be valuable for developers looking to optimize compile time effectively.

Reply:
Should be in the README.md within the `blocks` subdirectory and/or 'install & compile' description. I'd suggest to follow this up in another more documentation-related PR/issue.